pkg/git/libgit2/checkout.go

```diff
@@ -161,19 +161,44 @@ func (c *CheckoutBranch) Checkout(ctx context.Context, path, url string, opts *g
 		}
 		defer upstreamCommit.Free()
 
-		// Once the index has been updated with Fetch, and we know the tip commit,
-		// a hard reset can be used to align the local worktree with the remote branch's.
-		err = repo.ResetToCommit(upstreamCommit, git2go.ResetHard, &git2go.CheckoutOptions{
+		localBranch, err := repo.LookupBranch(c.Branch, git2go.BranchLocal)
```

Comment:
Can you also add a line above this describing why we are doing this? Just to keep a record of the knowledge.